src/resultsdisassemblypage.cpp

```diff
@@ -115,19 +124,52 @@ void ResultsDisassemblyPage::showDisassembly(QString processName, QStringList ar
 
         QStringList headerList;
         headerList.append(QLatin1String("Assembly"));
+        for (int i = 0; i < m_disasmResult.selfCosts.numTypes(); i++) {
+            headerList.append(m_disasmResult.selfCosts.typeName(i));
+        }
         model->setHorizontalHeaderLabels(headerList);
 
         QTextStream stream(&m_tmpFile);
         while (!stream.atEnd()) {
             QString asmLine = stream.readLine();
             if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly"))) continue;
 
+            QStringList asmTokens = asmLine.split(QLatin1Char(':'));
+            QString addrLine = asmTokens.value(0);
+            QString costLine = QString();
+
             QStandardItem *asmItem = new QStandardItem();
             asmItem->setText(asmLine);
             model->setItem(row, 0, asmItem);
+
+            // Calculate event times and add them in red to corresponding columns of the current disassembly row
+            for (int event = 0; event < m_disasmResult.selfCosts.numTypes(); event++) {
+                float totalCost = 0;
+                auto& entry = m_disasmResult.entry(m_curSymbol);
+                QHash<Data::Location, Data::LocationCost>::iterator i = entry.relSourceMap.begin();
+                while (i != entry.relSourceMap.end()) {
+                    Data::Location location = i.key();
+                    Data::LocationCost locationCost = i.value();
+                    float cost = locationCost.selfCost[event];
+                    if (QString::number(location.relAddr, 16) == addrLine.trimmed()) {
+                        costLine = QString::number(cost);
+                    }
+                    totalCost += cost;
+                    i++;
+                }
+
+                float costInstruction = costLine.toFloat();
+                costLine = costInstruction
+                    ? QString::number(costInstruction * 100 / totalCost, 'f', 2) + QLatin1String("%")
+                    : QString();
+
+                QStandardItem* costItem = new QStandardItem(costLine);
```

Comment:
you can setData(Qt::DisplayRole, costLine) (i.e. float -> QVariant), no need to convert to string here

Reply:
oh wait, you are trying to replicate what the real models are doing with qstandarditemmodel... sigh - as I said this really shouldn't be used and I hope you will clean this up in the future

anyhow, please still simplify the cost -> string -> float -> string conversion and only keep the final step

Reply:
and add a big fat FIXME that the QStandardItem stuff must be reimplemented properly